examples/transpose.jl

```diff
@@ -0,0 +1,55 @@
+using KernelAbstractions, CuArrays
+using StaticArrays
+
+@kernel function copy!(a,b)
+    i = @index(Global)
+    @inbounds b[i] = a[i]
+end
+
+@kernel function naive_transpose!(a,b, dim1, dim2)
+    cartesian_in = CartesianIndices(a)[@index(Global)]
+
+    # Changing index order for transpose.
+    # Note: the only way I can see to do this is to transfer it into an array
+    #       for a swap, and then transfer back
+    tmp_idx = MVector(cartesian_in.I)
+    @inbounds tmp_idx[dim1], tmp_idx[dim2] = tmp_idx[dim2], tmp_idx[dim1]
```

Comment:
We probably should provide a general method for doing `permutedims` on indexes.

Reply:
I think @mwarusz was right. This is an example. Let's keep it simple. No reason to do multi-dimensional distributed stuff.